pkgs/applications/networking/mailreaders/mailspring/default.nix

```diff
@@ -0,0 +1,85 @@
+{ stdenv
+, fetchurl
+, autoPatchelfHook
+, dpkg
+, glib
+, nss
+, gtk3
+, xorg
+, alsaLib
+, db
+, libkrb5
+, libsecret
+, udev
+, coreutils
+, openssl
+}:
+
+stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
+  pname = "mailspring";
+  version = "1.7.5";
+
+  src = fetchurl {
+    url = "https://github.com/Foundry376/Mailspring/releases/download/${version}/mailspring-${version}-amd64.deb";
+    sha256 = "07g614v6a9nwz7hm5vwfnc9lwjvnqvxgw2w5sj7di7b4irf69297";
+  };
+
+  nativeBuildInputs = [
+    dpkg
+    autoPatchelfHook
+  ];
+
+  buildInputs = [
+    glib
+    nss
+    gtk3
+    xorg.libXtst
+    xorg.libXScrnSaver
+    xorg.libxkbfile
+    alsaLib
+    db
+    libkrb5
+    libsecret
+  ];
+
+  runtimeDependencies = [
+    openssl
+    coreutils
+    udev.lib
+  ];
+
+  unpackPhase = /* sh */ ''
+    dpkg -x $src .
```

Comment:
I'm honestly not too familiar with policies on using dpkg (i.e. pre-compiled binaries) in Nix. I know general preference is to build from source if at all possible, but it seems this ISN'T possible due to a private C++ section of ``mailspring``? Is that correct? Did you try building from the GitHub at all?

Reply:
@drewrisinger thanks for your review.

Yes, mailspring depends on a private repo, that implements syncing mails. I found no other way to get the binary than depackaging one of the provided builds.